src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -285,5 +302,73 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryReadStatusFile(int pid, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            bool b = TryParseStatusFile(GetStatusFilePathForProcess(pid), out result, reusableReader);
+            Debug.Assert(!b || result.pid == pid, "Expected process ID from status file to match supplied pid");
+            return b;
+        }
+
+        private static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            string statusFileContents;
+            try
+            {
+                using (var source = new FileStream(statusFilePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
+                {
+                    statusFileContents = reusableReader.ReadAllText(source);
+                }
+            }
+            catch (IOException)
+            {
+                // Between the time that we get an ID and the time that we try to read the associated status
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var parser = new StringParser(statusFileContents, '\n');
+            var results = default(ParsedStatus);
+
+            var splits = new [] {':', ' ', '\t'};
+
+            while (parser.MoveNext())
+            {
+                var elements = parser.ExtractCurrent().Split (splits, 3, StringSplitOptions.RemoveEmptyEntries);
+                if (elements.Length < 2)
+                {
+                    continue;
+                }
+
+                switch (elements [0])
+                {
+                    case "Pid":
+                        results.pid = int.Parse (elements[1]);
```

Comment:
Is it possible this parsing will fail?  Wondering if it should be using TryParse instead for all of these.

Reply:
It shouldn't fail, but I'll use `TryParse` and return false in case it does.

Reply:
However, if we are to follow the same model as `TryParseStatFile`, we should fail if we can't parse the file successfully.